Dockerfile.ocp

```diff
@@ -0,0 +1,20 @@
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.14 AS builder
+
+COPY . /go/src/sigs.k8s.io/metrics-server
+
+WORKDIR /go/src/sigs.k8s.io/metrics-server
+RUN make metrics-server
```

Comment:
Maybe a `Makefile.ocp` that does `include Makefile` and has a suitably modified `metrics-server-ocp` target?

Then we at least only have to worry about changes in the `metrics-server` target, but not about merge conflicts.

Reply:
I was thinking if we can use this approach in syncbot https://github.com/rhobs/syncbot/blob/main/.github/workflows/merge-prometheus-adapter.yaml#L15-L16 so that we can keep downstream Makefile once we setup auto updates?

Either way is fine for me :)

Reply:
Restoring our downstream version and overwriting possible future upstream changes to the Makefile worries me a bit. This lead to strange failure that will be hard to debug.

Reply:
agree :)

Makefile.ocp

```
include Makefile

metrics-server-ocp: $(SRC_DEPS)
	GOARCH=$(ARCH) CGO_ENABLED=0 go build -mod=vendor -ldflags "$(LDFLAGS)" -o metrics-server sigs.k8s.io/metrics-server/cmd/metrics-server
```

on release repo PR we need to update respectively to use this Makefile